app/components/formatted_date/index.tsx

```diff
@@ -1,34 +1,38 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import moment from 'moment';
-import mtz from 'moment-timezone';
 import React from 'react';
 import {useIntl} from 'react-intl';
 import {Text, type TextProps} from 'react-native';
 
-import {getLocaleFromLanguage} from '@i18n';
+export type FormattedDateFormat = Exclude<Intl.DateTimeFormatOptions, 'timeZone'>;
 
 type FormattedDateProps = TextProps & {
-    format?: string;
+    format?: FormattedDateFormat;
     timezone?: string | UserTimezone | null;
     value: number | string | Date;
 }
 
-const FormattedDate = ({format = 'MMM DD, YYYY', timezone, value, ...props}: FormattedDateProps) => {
+const FormattedDate = ({
+    format = {dateStyle: 'medium'},
```

Comment:
Believe it or not, this creates a new object on every render if no object is passed.

It is not a big deal with how it is used in the component (not going into dependencies nor anything like that), but just to be future proof, do you mind defining it out of the component? Something like this:

```
const defaultFormat = {dateStyle: 'medium'};
const FormattedDate = ({
    format = defaultFormat,
    ...
```

Reply:
Sure !